build-tools/packages/build-cli/src/library/typescriptApi.ts

```diff
@@ -99,5 +117,73 @@ export function getApiExports(sourceFile: SourceFile): ExportRecords {
 		}
 	}
 
+	// If we have found levels for all things exported, then nothing else left to look for.
+	if (records.unknown.size === 0) {
+		return records;
+	}
+
+	// Otherwise, look for some special cases where level tagging may appear in source itself.
+
+	// Note that these are not exported declarations, but specifically
+	// given file's export declarations.
+	const exportDeclarations = sourceFile.getExportDeclarations();
+	for (const exportDeclaration of exportDeclarations) {
+		// console.log(`export @ ${exportDeclaration.getStartLineNumber()} has children:\n\t${exportDeclaration.getChildren().map((c) => c.getKindName()).join('\n\t')}`);
+
+		// Look first for namespace exports like `export * as foo`.
```

Comment:
Presumably this covers explicit `namespace` declarations as well?

Reply:
If a namespace is exported that should be covered by the logic below. If you have specific cases we can try 'em, out.
Tests are sorely lacking - especially when the results are not checked in.